TestApp/update-npm-packages.sh

```diff
@@ -13,32 +42,49 @@ npm pack ../appcenter-link-scripts
 echo "Installing appcenter* packages..."
 npm install appcenter*.tgz
 
-echo "workaround for macs on arm64"
-npm install react-native-fs --save --legacy-peer-deps
-
 echo "Cleanup appcenter*.tgz"
 rm appcenter*.tgz
 
 echo "Installing other packages..."
 npm install
 
-# Issue: https://github.com/react-native-community/cli/issues/1403
-# Fixed in RN 0.65: https://github.com/facebook/react-native/commit/8721ee0a6b10e5bc8a5a95809aaa7b25dd5a6043
-echo "Applying patch to react-native package that fixes build on Xcode 12.5..."
-npx patch-package
+if [ "$is_ios" == "true" ]; then
+    if [[ "$OSTYPE" == "darwin"* && "$(uname -m)" == "arm64" ]]; then
+        echo "Workaround for macs on arm64"
+        npm install react-native-fs --save --legacy-peer-deps
+    fi
+
+    echo "Updating CocoaPods repos..."
+    pod repo update
+
+    echo "Install shared framework pods..."
+    (cd ../AppCenterReactNativeShared/ios && pod install)
+
+    echo "Running pod install and building shared framework..."
+    (cd ios && pod install --repo-update)
+fi
 
-echo "Running jetify to resolve AndroidX compatibility issues..."
-npx jetify
 
-echo "Updating CocoaPods repos..."
-pod repo update
+if [ "$is_android" == "true" ]; then
+    echo "Running jetify to resolve AndroidX compatibility issues..."
+    npx jetify
 
-echo "Install shared framework pods..."
-(cd ../AppCenterReactNativeShared/ios && pod install)
+    OS_TYPE=$(uname)
+    if [[ "$OS_TYPE" == "Darwin" ]]; then
+        # macOS
+        SED_INPLACE=("sed" "-i" "")
+    else
+        # Linux and others
+        SED_INPLACE=("sed" "-i")
+    fi
 
-# for testing with not released apple and android sdks, you will need to provide the storage access key
-echo "Running pod install and building shared framework..."
-(cd ios && pod install --repo-update)
+    # Remove versionName and versionCode from build.gradle for AppCenterReactNativeShared
+    # to avoid build errors like:
+    # > Could not find method namespace() for arguments [com.microsoft.appcenter.reactnative.appcenter] on extension 'android' of type com.android.build.gradle.LibraryExtension.
```

Comment:
where does this error come from? Why is it happening? More explanation needed

Reply:
The part of the code that we are removing to prevent the error from occurring was added in this [PR](https://github.com/microsoft/appcenter-sdk-react-native/pull/1022) and is needed to work with the 1es release.

Reply:
PS: I fixed comments about which error appear.